core/src/com/unciv/logic/automation/unit/CivilianUnitAutomation.kt

```diff
@@ -117,6 +123,33 @@ object CivilianUnitAutomation {
                 UnitActions.invokeUnitAction(unit, UnitActionType.StartGoldenAge)
         }
 
+        if (unit.hasUnique(UniqueType.GainFreeBuildings)) {
+            val unique = unit.getMatchingUniques(UniqueType.GainFreeBuildings).first()
+            val buildingName = unique.params[0]
+            // Choose the city that is closest in distance and does not have the building constructed.
+            val cityToGainBuilding = unit.civ.cities.filter {
+                !it.cityConstructions.containsBuildingOrEquivalent(buildingName)
+                    && (unit.movement.canMoveTo(it.getCenterTile()) || unit.currentTile == it.getCenterTile())
+            }.minByOrNull {
+                val path = unit.movement.getShortestPath(it.getCenterTile())
+                path.size
+            }
+
+            if (cityToGainBuilding != null) {
+                if (unit.currentTile == cityToGainBuilding.getCenterTile()) {
+                    UniqueTriggerActivation.triggerUnique(unique, unit.civ, unit = unit, tile = unit.currentTile)
+                    UnitActionModifiers.activateSideEffects(unit, unique)
+                    return
+                }
+                else unit.movement.headTowards(cityToGainBuilding.getCenterTile())
+            }
+            return
+        }
+
+        // if none of the conditions above are met,
+        // AI will attempt to activate the first triggerable unique whenever it is available.
+        UnitActionsFromUniques.getTriggerUniqueActions(unit, unit.getTile()).firstOrNull()?.action?.invoke()
```

Comment:
Remove this - we should be implementing case-by-case, and this catchall trigger could lead to some really bad behaviour. I'd rather an AI doesn't know what it should do with a unit, than it activates its ability randomly
E.g. great generals that can create citadels - "create at first possible instance" is worse than "never create"

Reply:
I agree now; I don't want the unit to blindly remove buildings or turn tiles to something else